cdap-data-fabric/src/main/java/co/cask/cdap/data2/audit/AuditModule.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.audit;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.runtime.RuntimeModule;
+import com.google.inject.AbstractModule;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.name.Named;
+
+/**
+ * Guice bindings for publishing audit.
+ */
+public class AuditModule extends RuntimeModule {
+  @Override
+  public Module getInMemoryModules() {
+    return new AbstractModule() {
+      @Override
+      protected void configure() {
+        bind(AuditPublisher.class).to(NoOpAuditPublisher.class);
+      }
+    };
+  }
+
+  @Override
+  public Module getStandaloneModules() {
+    return new AbstractModule() {
+      @Override
+      protected void configure() {
+        bind(AuditPublisher.class).to(NoOpAuditPublisher.class);
+      }
+    };
+  }
+
+  @Override
+  public Module getDistributedModules() {
+    return new AbstractModule() {
+      @Override
+      protected void configure() {
+      }
+
+      @SuppressWarnings("unused")
+      @Provides
+      @Named(Constants.Audit.KAFKA_TOPIC)
+      public String providesKafkaTopic(CConfiguration cConf) {
```

Comment:
And have a `bind(AuditPublisher.class).toProvider(AuditPublisherProvider.class)` in the configure() method. The `CConfiguration` and `Injector` can be injected through the provider constructor and the `get()` method can return appropriate instance.